ocs_ci/ocs/utils.py

```diff
@@ -1003,6 +1048,46 @@ def collect_prometheus_metrics(
             json.dump(datapoints.json(), outfile)
 
 
+def save_live_logs(request, pods_containers_dict, patterns_to_log=None):
+    """
+    Save pods logs to files
+
+    """
+    tc_name = request.node.name
+
+    def finalizer():
+        """
+        Stop the thread that executed save_logs()
+
+        """
+        ocsci_config.RUN["save_live_pod_logs"] = False
+        if thread:
+            thread.join()
+
+    request.addfinalizer(finalizer)
+
+    ocsci_config.RUN["save_live_pod_logs"] = True
+
+    def save_logs():
+        while ocsci_config.RUN.get("save_live_pod_logs"):
```

Comment:
Just for me to learn... when should this for-loop stop? I mean when the condition of `ocsci_config.RUN.get("save_live_pod_logs")` will change to `False`?

Reply:
it changes to `False` and stops in https://github.com/red-hat-storage/ocs-ci/pull/4075/files#diff-13cfcd3566979733be4860f4cc0db5af5382acbf7cf54fd94eb36fab45474c21R1063

Which is the test teardown.
The idea of this new functionality is to save the logs of pods, every few seconds (10 seconds), as long as the test is running. In this PR, the test where I added this is `test_upgrade`, in order to collect Noobaa DB migration job pods, as in 4.7 Noobaa changes its DB from Mongo to PGSQL. 
The new capability that is being added in this PR allows collecting the logs of these DB migration job pods, which are getting deleted before the test ends so we don't have their logs collected in the must gather upon test completion

Reply:
Yes, I see. Thanks for the explanation.